pytorch_lightning/metrics/sklearn.py

```diff
@@ -0,0 +1,716 @@
+                Defaults to all processes (world)
+            reduce_op: the operation to perform during reduction within DDP (only needed for DDP training).
+                Defaults to sum.
+            **kwargs: additonal keyword arguments (will be forwarded to metric call)
+        """
+        super().__init__(name=metric_name, reduce_group=reduce_group,
+                         reduce_op=reduce_op)
+
+        self.metric_kwargs = kwargs
+        lightning_logger.debug(
+            f'Metric {self.__class__.__name__} is using Sklearn as backend, meaning that'
+            ' every metric call will cause a GPU synchronization, which may slow down your code'
+        )
+
+    @property
+    def metric_fn(self):
+        import sklearn.metrics
+        return getattr(sklearn.metrics, self.name)
+
+    def forward(self, *args, **kwargs) -> Union[np.ndarray, int, float]:
+        """
+        Carries the actual metric computation
+
+        Args:
+            *args: Positional arguments forwarded to metric call (should be already converted to numpy)
+            **kwargs: keyword arguments forwarded to metric call (should be already converted to numpy)
+
+        Return:
+            the metric value (will be converted to tensor by baseclass)
+
+        """
+        return self.metric_fn(*args, **kwargs, **self.metric_kwargs)
+
+
+class Accuracy(SklearnMetric):
+    """
+    Calculates the Accuracy Score
+
+    Warning:
+            Every metric call will cause a GPU synchronization, which may slow down your code
+    """
+    def __init__(self, normalize: bool = True,
+                 reduce_group: Any = torch.distributed.group.WORLD,
+                 reduce_op: Any = torch.distributed.ReduceOp.SUM):
+        """
+        Args:
+            normalize: If ``False``, return the number of correctly classified samples.
+                Otherwise, return the fraction of correctly classified samples.
+            reduce_group: the process group for DDP reduces (only needed for DDP training).
+                Defaults to all processes (world)
+            reduce_op: the operation to perform during reduction within DDP (only needed for DDP training).
+                Defaults to sum.
+        """
+        super().__init__(metric_name='accuracy_score',
+                         reduce_group=reduce_group,
+                         reduce_op=reduce_op,
+                         normalize=normalize)
+
+    def forward(self, y_pred: np.ndarray, y_true: np.ndarray,
+                sample_weight: Optional[np.ndarray] = None) -> float:
+        """
+        Computes the accuracy
+
+        Args:
+            y_pred: the array containing the predictions (already in categorical form)
+            y_true: the array containing the targets (in categorical form)
+            sample_weight:  Sample weights.
+
+        Return:
+            Accuracy Score
+
+        """
+        return super().forward(y_pred=y_pred, y_true=y_true, sample_weight=sample_weight)
```

Comment:
Do you need any of these forward functions in this file? It seems like you're implementing what inheritance will already do, i.e. call the superfunction if it's not overriden.

Reply:
@justusschock ^^

Reply:
@tullie you're right. But the superfunction takes args and kwargs as arguments. I have explicitly implemented these for documentation and type hinting purposes. But we can also remove them.